### PR TITLE
For release 2024.09.00

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
Updates `main` branch to contain commits for release `2024.09.00`

:construction: KNOWN ISSUES:construction: 
Layout transparency issue - Vector halo update bug, patch incoming
Bugs with high number of ranks when using `gt:cpu_ifirst` backend, patch incoming

### CODE REVIEW

File: examples/Fortran_serialization/01_serialize_fortran_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,667 @@
...
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "env: SERIALBOX_EXAMPLE_PATH=/home/ckung/Documents/Code/SMT-Nebulae-Tutorial/tutorial/Fortran_porting\n",
+      "env: SERIALBOX_INSTALL_PATH=/home/ckung/Documents/Code/SMT-Nebulae/sw_stack_path/install/serialbox/\n"
```
Comment (bensonr) [inline examples/Fortran_serialization/01_serialize_fortran_data.ipynb:68]: Are we okay with a specific user path being used in the example here - `ckung`?

File: examples/Fortran_serialization/01_serialize_fortran_data.ipynb [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "# Change SERIALBOX_EXAMPLE_PATH and SERIALBOX_INSTALL_PATH to appropriate paths\n",
+    "%env SERIALBOX_EXAMPLE_PATH=/home/ckung/Documents/Code/SMT-Nebulae-Tutorial/tutorial/Fortran_porting\n",
+    "%env SERIALBOX_INSTALL_PATH=/home/ckung/Documents/Code/SMT-Nebulae/sw_stack_path/install/serialbox/"
```
Comment (bensonr) [inline examples/Fortran_serialization/01_serialize_fortran_data.ipynb:75]: same comment regarding `ckung` references.

File: examples/Fortran_serialization/01_serialize_fortran_data.ipynb [hunk 3]
```diff
@@ -0,0 +1,667 @@
...
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "total 16\n",
+      "drwxrwxr-x 2 ckung ckung 4096 May 13 10:08 .\n",
+      "drwxrwxr-x 3 ckung ckung 4096 May 13 10:08 ..\n",
+      "-rw-rw-r-- 1 ckung ckung 5033 May 13 10:08 testSerialBox.F90\n"
```
Comment (bensonr) [inline examples/Fortran_serialization/01_serialize_fortran_data.ipynb:289]: Same comment regarding `ckung` references.

File: examples/Fortran_serialization/01_serialize_fortran_data.ipynb [hunk 4]
```diff
@@ -0,0 +1,667 @@
...
+      "-rw-rw-r-- 1 ckung ckung    872 May 13 10:08 ArchiveMetaData-FILLQ2ZERO_InOut.json\n",
+      "-rw-rw-r-- 1 ckung ckung    100 May 13 10:08 FILLQ2ZERO_InOut_fq_in.dat\n",
+      "-rw-rw-r-- 1 ckung ckung    100 May 13 10:08 FILLQ2ZERO_InOut_fq_out.dat\n",
+      "-rw-rw-r-- 1 ckung ckung    500 May 13 10:08 FILLQ2ZERO_InOut_m_in.dat\n",
+      "-rw-rw-r-- 1 ckung ckung    500 May 13 10:08 FILLQ2ZERO_InOut_m_out.dat\n",
+      "-rw-rw-r-- 1 ckung ckung    500 May 13 10:08 FILLQ2ZERO_InOut_q_in.dat\n",
+      "-rw-rw-r-- 1 ckung ckung    500 May 13 10:08 FILLQ2ZERO_InOut_q_out.dat\n",
+      "-rw-rw-r-- 1 ckung ckung   7157 May 13 10:08 MetaData-FILLQ2ZERO_InOut.json\n",
+      "-rwxrwxr-x 1 ckung ckung 997608 May 13 10:08 testSerialBox.bin\n",
+      "-rw-rw-r-- 1 ckung ckung   5033 May 13 10:08 testSerialBox.F90\n"
```
Comment (bensonr) [inline examples/Fortran_serialization/01_serialize_fortran_data.ipynb:416]: same comment regarding `ckung` references?

File: examples/Fortran_serialization/01_serialize_fortran_data.ipynb [hunk 5]
```diff
@@ -0,0 +1,667 @@
...
+      "-rw-rw-r-- 1 ckung ckung   6457 May 13 10:08 ArchiveMetaData-FILLQ2ZERO_InOut.json\n",
+      "-rw-rw-r-- 1 ckung ckung   1000 May 13 10:08 FILLQ2ZERO_InOut_fq_in.dat\n",
+      "-rw-rw-r-- 1 ckung ckung   1000 May 13 10:08 FILLQ2ZERO_InOut_fq_out.dat\n",
+      "-rw-rw-r-- 1 ckung ckung   5000 May 13 10:08 FILLQ2ZERO_InOut_m_in.dat\n",
+      "-rw-rw-r-- 1 ckung ckung   5000 May 13 10:08 FILLQ2ZERO_InOut_m_out.dat\n",
+      "-rw-rw-r-- 1 ckung ckung   5000 May 13 10:08 FILLQ2ZERO_InOut_q_in.dat\n",
+      "-rw-rw-r-- 1 ckung ckung   5000 May 13 10:08 FILLQ2ZERO_InOut_q_out.dat\n",
+      "-rw-rw-r-- 1 ckung ckung   9456 May 13 10:08 MetaData-FILLQ2ZERO_InOut.json\n",
+      "-rwxrwxr-x 1 ckung ckung 997648 May 13 10:08 testSerialBox_ts.bin\n",
+      "-rw-rw-r-- 1 ckung ckung   5117 May 13 10:08 testSerialBox_ts.F90\n"
```
Comment (bensonr) [inline examples/Fortran_serialization/01_serialize_fortran_data.ipynb:610]: same comment regarding `ckung` references.

**Review by bensonr — COMMENTED** [review]
There are many files now named `boilerplate`, I don't know what that means.

**Comment by FlorianDeconinck** [comment]
> There are many files now named `boilerplate`, I don't know what that means.

The boilerplate code that I wrote directly in NDSL is a one function system to pull a basic NDSL setup. It has been replicated in the notebooks, but we should merge those.

As for the notebooks listing `ckung`, since they are dealing with writing files it's difficult to not have some explicit folders in there. But when the users specifies `SERIALBOX_EXAMPLE_PATH` as asked in the begining of the notebook, they should see their own local machine.

**Comment by FlorianDeconinck** [comment]
Merged stop pending investigation into a `Mask` change to have side-effect on numerics

**Review by FlorianDeconinck — APPROVED** [review]

**Pull request merged**